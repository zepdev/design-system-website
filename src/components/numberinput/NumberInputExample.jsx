import React, { useState } from "react"
import NumberInput from "@zlab-de/zel-react/NumberInput"

function NumberInputExample() {
  const [count, setCount] = useState(0)

  return (
    <div>
      <NumberInput
        label="demo number input"
        count={count}
        onChange={value => setCount(value)}
        onMinus={() => setCount(count - 1)}
        onPlus={() => setCount(count + 1)}
      />
    </div>
  )
}

export default NumberInputExample
