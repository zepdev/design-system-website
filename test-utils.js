import React from "react"
import { render } from "@testing-library/react"
import { ThemeProvider } from "@material-ui/core/styles"
import theme from "./src/gatsby-theme-material-ui-top-layout/theme"

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
