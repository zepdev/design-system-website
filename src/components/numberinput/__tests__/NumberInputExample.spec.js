import React from "react"
import { render, fireEvent, screen } from "../../../../test-utils"
import NumberInputExample from "../NumberInputExample"

describe("NumberInputExample", () => {
  it("renders correctly", () => {
    const component = render(<NumberInputExample />)
    expect(component).toMatchSnapshot()
  })
  it("adds one", () => {
    render(<NumberInputExample />)
    fireEvent.click(screen.getAllByRole("button")[0])
    expect(screen.getByLabelText("demo number input").value).toBe("-1")
  })
  it("subtracts one", () => {
    render(<NumberInputExample />)
    fireEvent.click(screen.getAllByRole("button")[1])
    expect(screen.getByLabelText("demo number input").value).toBe("1")
  })
})
