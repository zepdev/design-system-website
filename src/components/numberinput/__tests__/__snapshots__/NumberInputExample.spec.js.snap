// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInputExample renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="zep-numberinput"
        >
          <button
            class="zep-button zep-button--icon"
          >
            <svg
              class="zep-button__icon"
              height="32"
              version="1.1"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Minus
              </title>
              <path
                d="M25.333 17.333h-18.667v-2.667h18.667z"
                fill="currentColor"
              />
            </svg>
          </button>
          <label
            class="zep-visually-hidden"
            for="numberInput_demo number input"
          >
            demo number input
          </label>
          <input
            class="zep-input zep-input--number"
            id="numberInput_demo number input"
            type="text"
            value="0"
          />
          <button
            class="zep-button zep-button--icon"
          >
            <svg
              class="zep-button__icon"
              height="32"
              version="1.1"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Plus
              </title>
              <path
                d="M25.333 17.333h-8v8h-2.667v-8h-8v-2.667h8v-8h2.667v8h8z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="zep-numberinput"
      >
        <button
          class="zep-button zep-button--icon"
        >
          <svg
            class="zep-button__icon"
            height="32"
            version="1.1"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Minus
            </title>
            <path
              d="M25.333 17.333h-18.667v-2.667h18.667z"
              fill="currentColor"
            />
          </svg>
        </button>
        <label
          class="zep-visually-hidden"
          for="numberInput_demo number input"
        >
          demo number input
        </label>
        <input
          class="zep-input zep-input--number"
          id="numberInput_demo number input"
          type="text"
          value="0"
        />
        <button
          class="zep-button zep-button--icon"
        >
          <svg
            class="zep-button__icon"
            height="32"
            version="1.1"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Plus
            </title>
            <path
              d="M25.333 17.333h-8v8h-2.667v-8h-8v-2.667h8v-8h2.667v8h8z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
